torchgeo/datasets/cabuar.py

```diff
@@ -0,0 +1,303 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""CaBuAr dataset."""
+
+import os
+from collections.abc import Callable
+from typing import ClassVar
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from matplotlib.figure import Figure
+from torch import Tensor
+
+from .errors import DatasetNotFoundError
+from .geo import NonGeoDataset
+from .utils import Path, download_url, lazy_import, percentile_normalization
+
+
+class CaBuAr(NonGeoDataset):
+    """CaBuAr dataset.
+
+    `CaBuAr <https://huggingface.co/datasets/DarthReca/california_burned_areas>`__
+    is a dataset for Change detection for Burned area Delineation and part of
+    the splits are used for the ChaBuD ECML-PKDD 2023 Discovery Challenge.
```

Comment:
Depending on how much code is the same, it may be possible to either subclass `ChaBuD`:
```python
from .chabud import ChaBuD

class CaBuAr(ChaBuD):
    ...
```
or create a shared abstract base class. We could either have both datasets in a single file or a different dataset in each file. It's borderline since it's more like a ChaBuD v2, but it has a unique name.

Reply:
Originally the data was proposed in CaBuAr, then we extended the dataset for the challenge. However, the challenge page is not working anymore.

Reply:
I can probably do the reverse if that seems appropriate: I can make ChaBuD an extension of CaBuAr since the latter deals already with more files and subsets.